apis/core/v1alpha1/function_types.go

```diff
@@ -24,97 +24,94 @@ import (
-	gr.HttpProxy = &httpProxy
-	gr.HttpsProxy = &httpsProxy
-	gr.NoProxy = &noProxy
-	gr.GitInitImage = &gitInitImage
-	gr.Submodules = &submodules
-	gr.SslVerify = &sslVerify
-	gr.Verbose = &verbose
-	gr.Depth = &depth
-}
-
-type Registry struct {
-	// Image registry url
-	Url *string `json:"url,omitempty"`
-	// Image registry account including username and password
-	Account *v1.SecretKeySelector `json:"account,omitempty"`
-}
-
-func (r *Registry) Init() {
-	var url string
-	r.Url = &url
-	r.Account = &v1.SecretKeySelector{}
+	gr.Credentials = &v1.LocalObjectReference{}
 }
 
 type Language string
 type Runtime string
 
 const (
 	Go            Language = "go"
-	Python        Language = "python"
 	Node          Language = "node"
 	BuildPhase             = "Build"
 	ServingPhase           = "Serving"
 	Created                = "Created"
-	Launching              = "Launching"
-	Launched               = "Launched"
+	Building               = "building"
+	Running                = "Running"
+	Succeeded              = "Succeeded"
 	Failed                 = "Failed"
+	Skipped                = "Skipped"
 	Knative       Runtime  = "Knative"
 	OpenFuncAsync Runtime  = "OpenFuncAsync"
+	Shipwright             = "Shipwright"
 )
 
+type Strategy struct {
+	// Name of the referent; More info: http://kubernetes.io/docs/user-guide/identifiers#names
+	Name string `json:"name"`
+
+	// BuildStrategyKind indicates the kind of the buildstrategy, BuildStrategy or ClusterBuildStrategy, default is BuildStrategy.
+	Kind *string `json:"kind,omitempty"`
+}
+
+type ShipwrightEngine struct {
+	// Strategy references the BuildStrategy to use to build the image.
+	// +optional
+	Strategy *Strategy `json:"strategy,omitempty"`
+	// Timeout defines the maximum amount of time the Build should take to execute.
+	//
+	// +optional
+	// +kubebuilder:validation:Format=duration
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+}
+
 type BuildImpl struct {
-	// Cloud Native Buildpacks builders
-	Builder string `json:"builder"`
-	// Environment params to pass to the builder
+	// Builder refers to the image containing the build tools inside which
+	// the source code would be built.
+	//
+	// +optional
+	Builder *string `json:"builder"`
+	// BuilderCredentials references a Secret that contains credentials to access
+	// the builder image repository.
+	//
+	// +optional
+	BuilderCredentials *v1.LocalObjectReference `json:"builderCredentials,omitempty"`
+	// The configuration for `Shipwright` build engine.
+	Shipwright *ShipwrightEngine `json:"shipwright,omitempty"`
+	// Params is a list of key/value that could be used to set strategy parameters.
+	// When using _params_, users should avoid:
+	// Defining a parameter name that doesn't match one of the `spec.parameters` defined in the `BuildStrategy`.
+	// Defining a parameter name that collides with the Shipwright reserved parameters. These are _BUILDER_IMAGE_,_DOCKERFILE_,_CONTEXT_DIR_ and any name starting with _shp-_.
```

Comment:
=>
	// Defining a parameter name that collides with the Shipwright reserved parameters including _BUILDER_IMAGE_,_DOCKERFILE_,_CONTEXT_DIR_ and any name starting with _shp-_.